### PR TITLE
Add CyclicLR scheduler

### PR DESCRIPTION
**Pull request by vincentamato**
## Proposed changes

- Implements cyclical learning rate with triangular wave patterns (CLR)
- Supports three scaling modes (`triangular`, `triangular2`, `exp_range`) and asymmetric cycles
- Includes comprehensive tests and documentation

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/ml-explore/mlx/blob/main/CONTRIBUTING.md) document
- [x] I have run `pre-commit run --all-files` to format my code / installed pre-commit prior to committing changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have updated the necessary documentation (if needed)

### CODE REVIEW

**Comment by angeloskath** [comment]
I feel that these are a bit niche (#2482 and #2483) and I am inclined to not merge them but maybe people can chime in if they think otherwise. Basically is anybody using anything other than cosine decay with warmup?

In addition, both of these can be implemented by joining simple schedules already present which is another reason not to merge them.

**Comment by vincentamato** [comment]
@angeloskath That makes sense. I was honestly just going through PyTorch docs and seeing if there was anything I could add to encourage parity. Out of curiosity, are there any features in PyTorch you'd like to bring over but haven't gotten around to? Would love to contribute in any way I can :)

**Comment by angeloskath** [comment]
>  Would love to contribute in any way I can

Thanks! We definitely appreciate that a lot! I don't think we should aim for parity because imho one of the strengths of MLX is that it still has significantly smaller surface area than PyTorch.

I think the most useful but arguably not particularly exciting would be to go over the issue list and grab any bugs and such.

Either way we still appreciate the contributions (even though we may not always merge them)!

**Comment by angeloskath** [comment]
Closing for the reasons mentioned above.

**Comment by angeloskath** [comment]
Closing for the reasons above.
